cmd/fluxd/main.go

```diff
@@ -334,23 +334,37 @@ func main() {
 
 	// Wrap the procedure for collecting images to scan
 	{
-		awsConf := registry.AWSRegistryConfig{
-			Regions:    *registryAWSRegions,
-			AccountIDs: *registryAWSAccountIDs,
-			BlockIDs:   *registryAWSBlockAccountIDs,
+		awsOptions := []string{
+			"registry-ecr-region",
+			"registry-ecr-include-id",
+			"registry-ecr-exclude-id",
 		}
-		credsWithAWSAuth, err := registry.ImageCredsWithAWSAuth(imageCreds, log.With(logger, "component", "aws"), awsConf)
-		if err != nil {
-			logger.Log("warning", "AWS authorization not used; pre-flight check failed")
-		} else {
-			imageCreds = credsWithAWSAuth
+		usingAWS := false
+		for _, awsOption := range awsOptions {
+			if fs.Changed(awsOption) {
```

Comment:
This breaks Weave Cloud and any other user that relies on Flux to automatically setup ECR auth.